src/pyodide.js

```diff
@@ -679,11 +679,43 @@ def temp(Module):
   globals = __main__.__dict__
   globals.update(builtins.__dict__)
 
+  global saved_globals
+  saved_globals = {}
+  saved_globals.update(globals)
```

Comment:
It's surprising to always make a copy of globals even when the user hasn't asked for it.

Couldn't we more explicitly do,
```py
state = Module.saveState()

# some time later

Module.resetState(state)
```
?

Reply:
Sure.